src/oca_github_bot/tasks/rebase_bot.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) ForgeFlow, S.L. 2020
+# Distributed under the MIT License (http://opensource.org/licenses/MIT).
+
+from .. import github
+from ..config import switchable
+from ..manifest import _user_can_push
+from ..process import CalledProcessError, check_call
+from ..queue import getLogger, task
+
+_logger = getLogger(__name__)
+
+
+@task()
+@switchable("rebase_bot")
+def rebase_bot_start(org, repo, pr, username, dry_run=False):
+    with github.login() as gh:
+        gh_pr = gh.pull_request(org, repo, pr)
+        target_branch = gh_pr.base.ref
+        try:
+            with github.temporary_clone(org, repo, target_branch) as clone_dir:
+                if not _user_can_push(
+                    gh, org, repo, username, clone_dir, target_branch
+                ):
```

Comment:
`user_can_push` is looking at modified addons in the current branch compared to `target_branch`. Here you have cloned `target_branch` there will no modified addons and `user_can_push` will always return false unless the user has write permissions on the repo.

So you need to checkout the PR branch.